janitor/functions/pivot.py

```diff
@@ -739,15 +757,14 @@ def _pivot_longer_names_pattern_sequence(
         len_index=len_index,
         reps=group_max,
         index=index,
-        outcome=outcome,
-        values=values,
+        outcome=values,
+        values=outcome,
```

Comment:
Is this intentional? Setting outcome=values and values=outcome

Reply:
I'll need to delve through the code again, it just caught my eye;  
because previously it was `values=values`, `outcome=outcome`, so was that a bug? I think I'm just missing why adding the `dropna` functionality would flip this relationship around -- I'm not seeing specifically which part of the newly introduced code would result in this specific behaviour.

Reply:
Sure, I'll rework d code and add comments. Hopefully I don't make it worse 🤣

Reply:
@thatlittleboy I renamed the variables; the logic remains the same. Kindly have a look when you can, and let me know where it still has kinks. thanks